### ISSUE TITLE
Request Header 'Content-Type' no longer set for requests with body serialized as FormData 

### ISSUE THREAD

**Issue by schuch** [drwpow/openapi-typescript#1548]
**Description**

After upgrading `openapi-fetch` from 0.8.2 to 0.9.1 we see `415 Unsupported Media Type` responses for multipart/form-data requests.

We have investigated how the request has changed as a result of the update. We noticed that the request header `Content-Type: multipart/form-data; boundary=...` is no longer sent with the request.

A copy from of the request (as cUrl) from the developer console:

```
curl 'https://...' \
  -H 'Accept: */*' \
  --data-raw $'------WebKitFormBoundary9Y0HMzSGLEhBsYQT\r\nContent-Disposition: form-data; name="file"; filename="test.pdf"\r\nContent-Type: application/pdf\r\n\r\n\r\n------WebKitFormBoundary9Y0HMzSGLEhBsYQT--\r\n' \
  --compressed
```

(omitted a few headers for better clarity)

**Reproduction**

We use the following code to perform the request:

```ts
await POST(`/...`, {
    params: {
      ...
    },
    body: {
      file
    },
    bodySerializer: (body) => {
      const formData = new FormData();
      for (const name in body) {
        formData.append(name, body[name as keyof typeof body]);
      }
      return formData;
    }
).then(...)
```

**Expected result**

openapi-fetch create a proper multipart form-data request, e.g.:

```
curl 'https://...' \
  -H 'Accept: */*' \
  -H 'Content-Type: multipart/form-data; boundary=----WebKitFormBoundaryy81TvwmCQwmGHDDw' \
  --data-raw $'------WebKitFormBoundaryy81TvwmCQwmGHDDw\r\nContent-Disposition: form-data; name="file"; filename="test.pdf"\r\nContent-Type: application/pdf\r\n\r\n\r\n------WebKitFormBoundaryy81TvwmCQwmGHDDw--\r\n' \
  --compressed
```
(copied as cUrl from a successful request from the developer console, omitted a few headers for better clarity)

**Checklist**

- [x] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-typescript/blob/main/packages/openapi-fetch/CONTRIBUTING.md))

**Comment by shirish87** [comment]
I have the same issue. Inspecting the `Request` object in openapi-fetch shows the `content-type` header that used to be set in 0.8.2  `multipart/form-data; boundary=----formdata-undici-0.6228903411172271` is no longer set in 0.9.x

**Comment by lukasbash** [comment]
Is there a chance to get this in a patch release as soon as merged? Would be awesome!

**Comment by drwpow** [comment]
Released in `0.9.2`!
